storage/txcache/factory.go

```diff
@@ -0,0 +1,13 @@
+package txcache
+
+// CreateCache creates a (fail-safe) transactions cache
+func CreateCache(config CacheConfig) txCache {
+	cache, err := NewTxCache(config)
+	if err != nil {
+		log.Error("CreateCache()", "err", err)
```

Comment:
Same question. Why is this a logged error? It can happen during normal operation? If not, why not returning the error? Why does the program runs as nothing bad had happened?

Reply:
Good point! I link here the comments from previous PRs:

[1767](https://github.com/ElrondNetwork/elrond-go/pull/1767#discussion_r429854001)
[1716](https://github.com/ElrondNetwork/elrond-go/pull/1716#discussion_r424249173)

Firstly - it seems that the factory function wasn't used by `shardedTxPool` - possible merge issue, thanks for pointing this out - fixed.

Secondly (comment from previous PR): we should change `shardedTxPool` to create all the needed caches in its constructor, and this way we can return the critical error and stop the node. We can apply this change in the upcoming PR that separates `TxCache` from `CrossTxCache`. And this means we might remove this disabled cache component in the next PR. We have this TODO: _Perhaps create all caches in constructor?_

Is this ok?

Reply:
Ok. Will remain as it is for now.